File: cloudflow-frontend/src/services/api/dutySchedule.ts
import request from './request';
import { PageResult } from '@/types';

/** 值班排班接口类型 */
export interface DutySchedule {
  scheduleId?: number;
  tenantId?: number;
  title: string;
  scheduleType: string;     // DAILY / HOLIDAY / EMERGENCY
  dutyDate: string;
  shiftType?: string;       // DAY / NIGHT / FULL
  startTime?: string;
  endTime?: string;
  userId: number;
  userName?: string;
  backupUserId?: number;
  backupUserName?: string;
  deptId?: number;
  deptName?: string;
  location?: string;
  dutyContent?: string;
  checkInTime?: string;
  checkOutTime?: string;
  status?: string;
  swapReason?: string;
  remark?: string;
  createTime?: string;
}

/** 值班排班 API */
export const dutyScheduleApi = {
  list: (params: { pageNum?: number; pageSize?: number; status?: string; scheduleType?: string; userId?: number }) =>
    request.get('/oa/duty/list', { params }) as Promise<PageResult<DutySchedule>>,
  calendar: (params: { year: number; month: number; deptId?: number }) =>
    request.get('/oa/duty/calendar', { params }) as Promise<DutySchedule[]>,
  getInfo: (id: number) => request.get(`/oa/duty/${id}`) as Promise<DutySchedule>,
  add: (data: DutySchedule) => request.post('/oa/duty', data),
  addBatch: (data: DutySchedule[]) => request.post('/oa/duty/batch', data),
  edit: (data: DutySchedule) => request.put('/oa/duty', data),
  remove: (ids: number[]) => request.delete(`/oa/duty/${ids.join(',')}`),
  checkIn: (id: number) => request.put(`/oa/duty/checkin/${id}`),
  checkOut: (id: number) => request.put(`/oa/duty/checkout/${id}`),
  swap: (id: number, data: { backupUserId: number; backupUserName: string; reason: string }) =>
    request.put(`/oa/duty/swap/${id}`, data),
};

File: cloudflow-backend/cloudflow-service-oa/src/main/java/com/cloudflow/oa/domain/DutySchedule.java
package com.cloudflow.oa.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 值班排班实体类
 * 对应表：sys_duty_schedule
 */
@Data
@TableName("sys_duty_schedule")
public class DutySchedule implements Serializable {
    private static final long serialVersionUID = 1L;

    /** 排班ID */
    @TableId(value = "schedule_id", type = IdType.AUTO)
    private Long scheduleId;

    /** 租户ID */
    private Long tenantId;

    /** 排班标题 */
    private String title;

    /** 排班类型(DAILY日常值班/HOLIDAY节假日值班/EMERGENCY应急值班) */
    private String scheduleType;

    /** 值班日期 */
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDateTime dutyDate;

    /** 班次(DAY白班/NIGHT夜班/FULL全天) */
    private String shiftType;

    /** 值班开始时间 */
    private String startTime;

    /** 值班结束时间 */
    private String endTime;

    /** 值班人ID */
    private Long userId;

    /** 值班人姓名 */
    private String userName;

    /** 替班人ID */
    private Long backupUserId;

    /** 替班人姓名 */
    private String backupUserName;

    /** 部门ID */
    private Long deptId;

    /** 部门名称 */
    private String deptName;

    /** 值班地点 */
    private String location;

    /** 值班内容/职责 */
    private String dutyContent;

    /** 签到时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime checkInTime;

    /** 签退时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime checkOutTime;

    /** 状态(SCHEDULED/CHECKED_IN/COMPLETED/SWAPPED/CANCELLED) */
    private String status;

    /** 换班原因 */
    private String swapReason;

    /** 备注 */
    private String remark;

    /** 删除标志 */
    private String delFlag;

    /** 创建者 */
    private String createBy;

    /** 创建时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    /** 更新者 */
    private String updateBy;

    /** 更新时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
}
